rswag-specs/lib/rswag/specs/request_factory.rb

```diff
@@ -118,6 +118,21 @@ def add_path(request, metadata, swagger_doc, parameters, example)
 
       def build_query_string_part(param, value)
         name = param[:name]
+
+        # OAS 3: https://swagger.io/docs/specification/serialization/
+        if param.has_key? :schema
+          return "#{name}=#{value}" unless param[:schema][:type].to_sym == :object
+
+          style = param[:style] || :form
```

Comment:
Thinking about it, as per the pattern set by `param[:schema][:type]` I think it'd be nice to allow `style: 'deepObject'` to work here too by doing something like:

```suggestion
          style = (param[:style] || :form).to_sym
```